arkouda/join.py

```diff
@@ -205,6 +210,33 @@ def inner_join(
     """
     from inspect import signature
 
+    if type(left) != type(right):
+        raise TypeError("Left and Right arrays must be of same type")
+
+    # Convert Strings and Categorical left & right values to Categorical.codes. This generates an
+    # index representation of the strings. This way they can be handled as numerics and follow the
+    # same process as int
+    t = cast(str, pdarray.objtype)
+    cats = None
+
+    if isinstance(left, Strings) and isinstance(right, Strings):
+        left = Categorical(left)
+        right = Categorical(right)
+
+    if isinstance(left, Categorical) and isinstance(right, Categorical):
+        t = cast(str, Categorical.objtype)
+        left, right = Categorical.standardize_categories([left, right])
+        if isinstance(left, Categorical) and isinstance(right, Categorical):
+            cats = left.categories[left.codes]  # Get standardized categories without NAvalue
+            leftKeepCodes = array(
+                [x is not True for x in left.isna().to_ndarray()]
+            )  # Keep codes that aren't NAvalue
+            rightKeepCodes = array(
+                [x is not True for x in right.isna().to_ndarray()]
+            )  # Keep codes that aren't NAvalue
+            left = left.codes[leftKeepCodes]
+            right = right.codes[rightKeepCodes]
```

Comment:
since we call `to_ndarray` it's still possible to exceed max_transfer bytes. In this example I'm artificially lower the max, but this could definitely happen for a reasonably sized strings array if every element is unique
```python
In [3]: int_left = ak.arange(100)
   ...: int_right = ak.arange(100)
   ...: str_left = ak.array([f'str {i}' for i in int_left.to_list()])
   ...: str_right = ak.array([f'str {i}' for i in int_right.to_list()])

In [4]: ak.client.maxTransferBytes = 10

In [5]: sl, sr = ak.join.inner_join(str_left, str_right)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[5], line 1
----> 1 sl, sr = ak.join.inner_join(str_left, str_right)

RuntimeError: Array exceeds allowed size for transfer. Increase client.maxTransferBytes to allow
```

Reply:
Replaced the list interpretation doing `to_ndarray()`